src/main/java/com/google/maps/DirectionsClientFactory.java

```diff
@@ -14,7 +14,16 @@
 
 package com.google.maps;
 
-/** Contract for creating a DirectionsClient with a given API key. */
+/**
+ * Contract for creating a DirectionsClient with a given API key. Implement getDirectionsClient to
+ * obtain an instance of DirectionsClient.
+ */
 public interface DirectionsClientFactory {
+  /**
+   * Gets a DirectionsClient which executes against the given API key.
+   *
+   * @param apiKey A string representing the API key to authenticate a Google Directions API call.
+   * @return DirectionsClientImpl instance which executes against the given API key.
+   */
   DirectionsClient getDirectionsClient(String apiKey);
```

Comment:
I think I would prefer to keep the pattern and perhaps in the future, make the factory obtain the `apiKey` instead of the servlet. Also, I am considering re-purposing this to be a MapsClient which handles calls to and from Maps APIs such as Places and Directions. Let me know your thoughts!

Reply:
My original comment was about having the following pair of constructor `DirectionsClientImpl(String apiKey)` and `DirectionsServlet(DirectionsClient client)` so you wouldn't need the factory at all, but you can it as is, I don't have a strong opinion on that.

Having a single interface handle Places and Directions seems good!

Reply:
Oooh okay I see what you mean now. I'll definitely keep that in mind when refactoring and repurposing this then!

Reply:
I have created issue #95 which includes this consideration when I get to refactoring Directions client into a more general Maps client.